nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/ml/AwsMlJobStarter.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+        RESPONSE response;
+        FlowFile childFlowFile;
+        try {
+            response = sendRequest(buildRequest(session, context, flowFile), context);
+            childFlowFile = writeToFlowFile(session, flowFile, response);
+            postProcessFlowFile(context, session, childFlowFile, response);
+        } catch (Exception e) {
+            session.transfer(flowFile, REL_FAILURE);
+            getLogger().error("Sending AWS ML Request failed", e);
+            return;
+        }
+        session.transfer(flowFile, REL_ORIGINAL);
+        session.transfer(childFlowFile, REL_SUCCESS);
+    }
+
+    protected void postProcessFlowFile(ProcessContext context, ProcessSession session, FlowFile flowFile, RESPONSE response) {
+        session.putAttribute(flowFile, AWS_TASK_ID_PROPERTY, getAwsTaskId(context, response));
+        getLogger().debug("AWS ML task has been started with task id: {}", getAwsTaskId(context, response));
+    }
+
+    protected REQUEST buildRequest(ProcessSession session, ProcessContext context, FlowFile flowFile) throws JsonProcessingException {
+        return mapper.readValue(getPayload(session, context, flowFile), getAwsRequestClass(context));
+    }
+
+    private String getPayload(ProcessSession session, ProcessContext context, FlowFile flowFile) {
+        String payloadPropertyValue = context.getProperty(JSON_PAYLOAD).evaluateAttributeExpressions(flowFile).getValue();
+        if (payloadPropertyValue == null) {
+            payloadPropertyValue = readFlowFile(session, flowFile);
+        }
+        return payloadPropertyValue;
+    }
+
+    @Override
+    protected SERVICE createClient(ProcessContext context, AWSCredentials credentials, ClientConfiguration config) {
+        throw new UnsupportedOperationException("Tried to create client in a deprecated way.");
+    }
+
+    abstract protected RESPONSE sendRequest(REQUEST request, ProcessContext context) throws JsonProcessingException;
+
+    abstract protected Class<? extends REQUEST> getAwsRequestClass(ProcessContext context);
+
+    abstract protected String getAwsTaskId(ProcessContext context, RESPONSE response);
+
+    protected FlowFile writeToFlowFile(ProcessSession session, FlowFile flowFile, RESPONSE response) {
+        FlowFile childFlowFile = session.create(flowFile);
+        session.write(childFlowFile, out -> {
+            try (BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(out, StandardCharsets.UTF_8))) {
+                bufferedWriter.write(mapper.writeValueAsString(response));
+                bufferedWriter.newLine();
+                bufferedWriter.flush();
+            }
```

Comment:
See note on using `writeValueAsBytes()` instead of the BufferedWriter and String approach.